source/MenuPanel.cpp

```diff
@@ -89,7 +89,16 @@ void MenuPanel::Step()
 	{
 		++scroll;
 		if(scroll >= (20 * credits.size() + 300) * scrollSpeed)
+		{
+			creditIndex = (creditIndex + 1) % GameData::Sources().size();
 			scroll = 0;
+			string source = GameData::Sources()[creditIndex];
+			size_t pos = source.rfind('/', source.length() - 2) + 1;
+			string name = source.substr(pos, source.length() - 1 - pos);
+			credits.clear();
+			if(Files::Exists(GameData::Sources()[creditIndex] + "credits.txt"))
+				credits = Format::Split(Files::Read(source + "credits.txt"), "\n");
+		}
```

Comment:
So once we reach the end of the scroll, we're incrementing to the next source to show the credits from. But if the next source doesn't have a credits file, we're just scrolling through an empty credits list until the scroll is greater than the scroll speed * 300? Should we instead skip over sources that don't have credits? Otherwise, I believe someone with a lot of plugins that don't have credits (which one would imagine could be a lot since existing plugins likely don't have credits) might get a lot of empty space after the vanilla credits are done.